src/main/kotlin/com/sourcegraph/cody/chat/AgentChatSession.kt

```diff
@@ -37,25 +40,28 @@ private constructor(
    */
   private val sessionId: AtomicReference<CompletableFuture<SessionId>> =
       AtomicReference(newSessionId)
-  private val chatPanel: ChatPanel = ChatPanel(project, this)
+  private val chatPanel: ChatPanel = ChatPanel(project, this, selectedModel)
   private val cancellationToken = AtomicReference(CancellationToken())
   private val messages = mutableListOf<ChatMessage>()
 
   init {
     cancellationToken.get().dispose()
+    sessionId.get().thenAccept { sessionId ->
+      chatPanel.addModelDropdown(project, sessionId, selectedModel)
+    }
   }
 
   fun restoreAgentSession(agent: CodyAgent) {
-    // todo serialize model
-    val model = "anthropic/claude-2.0"
     val messagesToReload =
         messages
             .toList()
             .dropWhile { it.speaker == Speaker.ASSISTANT }
             .fold(emptyList<ChatMessage>()) { acc, msg ->
               if (acc.lastOrNull()?.speaker == msg.speaker) acc else acc.plus(msg)
             }
-    val restoreParams = ChatRestoreParams(model, messagesToReload, UUID.randomUUID().toString())
+    val restoreParams =
+        ChatRestoreParams(
+            selectedModel?.agentName ?: "", messagesToReload, UUID.randomUUID().toString())
```

Comment:
Is `""` really correct, shouldn't it be null to use the default? Have you tested this path?